ws-nextjs-app/pages/[service]/send/[id]/FormField/File/index.tsx

```diff
@@ -125,11 +143,16 @@ export default ({ id, name, inputState, describedBy, label }: InputProps) => {
     ) as File[];
     const uploaded = [...filesInState];
 
+    // Needs translation
+    let liveRegionText = `Update, Here's what you're sending: `;
     chosenFiles.forEach(file => {
       uploaded.push(file);
+      liveRegionText = `${liveRegionText}${file.name}, `;
     });
 
     handleChange(name, uploaded);
+    // Adds 1s delay for MacOS file explorer to bring focus back to browser, so VoiceOver can pickup the DOM change in time
+    setTimeout(() => replaceLiveRegionWith(liveRegionText), 1000);
```

Comment:
Does the `setTimeout` need any clean-up when the component unmounts?

Reply:
I don't think so as it's not contained in a useEffect hook, might be wrong there though

Reply:
Probably not a huge issue tbh. Would be more so in a SPA as I think setTimeout will get added to the call stack and when the component unmounts the setTimeout will still be in memory and execute. But in our case, if the user navigates away or to another page it will be a full page reload, so memory will be flushed.

Reply:
I guess it could get troublesome if the user spams through file adding, hopefully not enough to break the page. How do we cleanup the timers in memory with arrow functions? Or do we have to convert it to normal function declaration to access `componentWillUnmount()`

Reply:
Actually thinking about it, this could be a problem. If a timeout is running and the user submits within 1 second, the component would've unmounted and mounted the uploading state, and then the timeout tries to update state in LiveRegionContextProvider, which won't exist causing an error?

Reply:
Ahh I think we can use a ref for the timer, then useEffect to do the cleanup

Reply:
Yea potentially actually. What you can do is use a `useRef` and then store the `setTimeout` against that ref. Then have a `useEffect` that just has a cleanup function on unmount like:
```
  const timeoutRef = useRef(null);
  ...
  timeoutRef.current = setTimeout(....)
  ....
  useEffect(() => {
    return () => {
      clearTimeout(timeoutRef.current)
    }
  }, []);
```

Reply:
doooone